test/integration/preflight/preflight_test.go

```diff
@@ -2,11 +2,39 @@ package preflight
 
 import (
 	"bytes"
+	"flag"
+	"fmt"
+	"os"
 	"os/exec"
+	"sync"
 	"syscall"
 	"testing"
 )
 
+var prebuildPlays = []string{
+	"check-yum-update/build.yml",
+}
+
+func runPlaybook(file string, wg *sync.WaitGroup) {
+	cmd := exec.Command("ansible-playbook", file)
+	if output, err := cmd.Output(); err != nil {
+		fmt.Printf("Pre-run build laybook failed: %s\n%s", file, output)
+		os.Exit(1)
+	}
+	wg.Done()
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	wg := new(sync.WaitGroup)
+	wg.Add(len(prebuildPlays))
+	for _, file := range prebuildPlays {
+		go runPlaybook(file, wg)
+	}
```

Comment:
Don't share the `WaitGroup`, that easily leads to synchronization problems. Imagine that after changing `runPlaybook` we forget to call `wg.Done()` in one code path, the `wg.Wait()` below would block forever.

We can use `wg` in a single goroutine (the main goroutine):

```go
func TestMain(m *testing.M) {
	flag.Parse()
	var wg sync.WaitGroup // just need to declare it, and it is ready to use
	wg.Add(len(prebuildPlays))
	for _, file := range prebuildPlays {
		go func(file string) {
			runPlaybook(file)
			wg.Done()
		}(file)
	}
	wg.Wait()
	os.Exit(m.Run())
}

func runPlaybook(file string) {
	cmd := exec.Command("ansible-playbook", file)
	if output, err := cmd.Output(); err != nil {
		fmt.Printf("Pre-run build laybook failed: %s\n%s", file, output)
		os.Exit(1)
	}
}
```

And perhaps this is coding for the future, we could start with just running the one playbook we have:

```go
func TestMain(m *testing.M) {
	flag.Parse()
	cmd := exec.Command("ansible-playbook", "check-yum-update/build.yml")
	if b, err := cmd.CombinedOutput(); err != nil {
		fmt.Fprintf(os.Stderr, "Pre-run build playbook failed: %v\nOutput:\n%s", err, b)
		os.Exit(1)
	}
	os.Exit(m.Run())
}
```

And bonus points to running `ansible-playbook` with colors :smile: 
[test/integration/preflight/preflight_test.go#L29](https://github.com/rhcarvalho/openshift-ansible/blob/7be36890ba958a59cb3e0b851d81296058532a2f/test/integration/preflight/preflight_test.go#L29)